Create builder for TransitDataProviderFilter

habrahamsson-skanetrafiken opened the pull request:
### Summary

This PR is a small refactor of the `RouteRequestTransitDataProviderFilter`. It doesn't change any behavior at all.

- Adds a builder to simplify using this in places where we don't have a `RouteRequest`.
- Renames the filter to `DefaultTransitDataProviderFilter`.

### Unit tests

No new unit tests are added, but the builder makes a lot of tests much more readable.

### Documentation

None needed IMO

### Bumping the serialization version id

Nope

t2gran commented on application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/RouteRequestTransitDataProviderFilter.java:
```diff
@@ -109,13 +120,11 @@ private boolean tripTimesPredicate(TripTimes tripTimes, boolean applyTripTimesFi
 
-    if (wheelchairEnabled) {
-      if (
-        wheelchairPreferences.trip().onlyConsiderAccessible() &&
```
This is removed: `wheelchairPreferences.trip().onlyConsiderAccessible()` intensional?

habrahamsson-skanetrafiken commented:
Edit: Changed the commit messages to use "refactor:" instead of "[Refactor]".

jessicaKoehnke reviewed (approved):
Nice simplification!

habrahamsson-skanetrafiken commented:
I noticed a slight change in behavior. An explicitly empty filter list in the request would originally disallow all transit. This behavior is undocumented and differs from what we usually do. In order to not break any clients in the event that they are depending on this I fixed it and added a test.
